### DISCUSSION TITLE
[Feature] path-based-image-id

### DISCUSSION THREAD

**Discussion by andyxpert** [immich-app/immich#13157] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

I had a problem 2 days ago - external library HDD got disconnected and the immich nightly job "forgot" the photos on the HDD. Nothing I did helped so without a DB backup I had to readd the external library. Now I hope in about 1 week it finishes thumbnail regeneration (even though the paths are the same, the images and thumbs are the same... Immich needs to recreate them because the ids are lost)

I gave up hoping for a faster thumbnail generation, but what if the image ids are reversible hashes made from the image path ?!? 
They will surely be unique, they could be unhashed to verify if the image is the same (to avoid regenerating the thumbnail) and it would be a better, scalable and additive solution than the existing (md5 maybe ?) hashing mechanism.

What do you think ? Would this help speed up lengthy regeneration of already existing images ?
Or simply keep in the db table the image path (which i think already exists) and alter the code to check the path before regenerating a thumbnail for an image that was already parsed.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by andyxpert** [comment]
If there is a way to avoid thumb regeneration in such a scenario, please let me know. At least for any future such situation or any similar situation where Immich mass removes many images from external libraries.
I understand that I'm in a situation where to Immich it looks like I removed all the images and it (correctly) assumes that it has to remove them from its DB... But maybe it can keep them trashed for 30 days and if they reappear it can simply untrash them (manually even) ?!?

I'm saying this because the thumbs were not automatically removed... They got removed when I removed the library and added it back and it started generating the thumbs... So maybe it could have reused them instead of regenerating them...

**Comment by zackpollard** [comment]
Hey, when external libraries get disconnected, the assets get set to "offline". We don't actually remove the assets from the server for 30 days by default, however you may not have realised this is what happened. The assets were likely still there, but would've been showing in the trash view. On 1.116 there was a bug where re-running the scan with offline assets wouldn't bring the assets back online, however this  has been corrected in 1.117. This is therefore likely just a series of unfortunate events for you.

**Reply by andyxpert** [reply]
I was just reading the release notes for 117... damn, indeed a series of unfortunate events indeed... 
Had I just waited a few days it would all have been ok...

Disregard most of my proposals as after reading through the documentation I saw that you do indeed trash for 30 days, and now with the ver 117 improvements on thumb generation process things should be slightly smoother.

Well now, keep up the great work guys and I hope you'll add an API that allows an array of image paths to an album (now I think I can only add images to albums by id, so I have to grab ids with API search by image path and then pass them to add to album... 
If you have a script that does this or a faster/better way let me know pls, otherwise I'll create some bash for this...
